### DISCUSSION TITLE
Commonly used responses system/manager - Similar to Answer Files

### DISCUSSION THREAD

**Discussion by vvincevv** [NUWCDIVNPT/stig-manager#1048] [Ideas and Enhancements]
Some type of response system/manager that we would be able to enter commonly used finding details/comments for different STIG findings and then be able to recall them for use throughout the application.

The settings would possibly be under Application Management or a new heading.

You would start off by selecting a STIG and the a STIG Id/Rule ID/Group ID and then be able to type in what you want to be able to populate as the Result/Finding Details/Comment.

We would then be able to choose these presets to insert when editing either a single STIG or batch edits.

This would of course be similar to the Answer File functionality in Evaluate-STIG, but would be able to be used by all STIGs, not just what is supported by Evaluate-STIG. Also, since it would be centrally managed in STIG Manager, rather than in multiple answer files throughout different environments it would allow us to audit the templates/answers much easier for approval/verification. Especially if it kept track of the last time it was edited/approved for use.

Since this is comparable to Evaluate-STIG answer files, I don't know if there should be functionality where you can set If [status] do this, because in bulk edits there are cases where a finding may be NA for 1(or more) servers so the template should only apply if it is currently Open, such as if it only applies to non domain joined or only DC's.

**Comment by cd-rite** [comment]
Hi @vvincevv 
Thanks for the suggestion! We will consider whether and where it would be appropriate to offer a feature like this. In general, we avoid any "automatic" population of answers that can't be traced back to specific imports or user evaluation. 
Also, offering a feature like this at the "Application Management" level might not be appropriate, as I suspect it's possible different Collections would have different acceptable answers.

If you just want to make sure people have an example of a good answer for their Reviews, we might already have something that could work for you:

To get a similar function with our current feature set, you could create an "example" asset in a Collection with Reviews that you consider appropriate, and then use our **"Other Assets" drag & drop feature** to populate the Asset under Review with the same answers as appropriate. If you name it something like "-Asset-with-good-answers-!" it will always be at the top of that list. You could also give that Asset a distinctive label to help it stand out. **Hovering over it will show the detail/comment, and from there you can drag and drop it into the Evaluation panel for the Asset being Reviewed,** then modify/save that Asset's Review:
![image](https://github.com/NUWCDIVNPT/stig-manager/assets/61710958/10407f74-13cf-4fb7-995b-3edc26c9ecb9)

![image](https://github.com/NUWCDIVNPT/stig-manager/assets/61710958/c716bc26-e320-4673-b511-0099e8d6c592)


Naming it like that would also make sure it appears at the top of the Collection Review workspace as well, though it would require a cut and paste operation rather than drag and drop. 

Hope that helps!

**Reply by jace5869** [reply]
Is this something that is still functioning? I can not seem to make this work. 
Dragging does show the text, but it doesn't populate answers to the other assets you drop it on.
Also looks like you can not select multiple assets to modiy.

**Reply by cd-rite** [reply]
Hi @jace5869 The above applies to the individual Asset Review interface, dragging from the "Other Assets" panel onto an empty or "saved" Review (not submitted/accepted).  
It sounds like you're talking about the "Collection Review" interface. That one does not support drag and drop, but has a "Batch Edit" interface that will let you set the result/detail/comment for any assets you have selected.

**Comment by dtester1** [comment]
Wouldn't bulk edit be an option for this?  Create one answer for the STIG check and apply it to all assets.  This way the check was being reviewed while this was being done.  Really this is only needed for manual stigs at this point with the evolution of evaluate-stig.  Also trying in general to figure out where canned answers are applicable as we aren't supposed to use answers that are the same for items like mitigations and such.  Maybe I am missing something though here.
